library/src/plan.cpp

```diff
@@ -1007,21 +1007,41 @@ bool TreeNode::use_CS_2D_SINGLE()
 
 bool TreeNode::use_CS_2D_RC()
 {
-    //   For CS_2D_RC, we are reusing SBCC kernel for 1D middle size. The
-    //   current implementation of 1D SBCC supports only 64, 128, and 256.
-    //   However, technically no LDS limitation along the fast dimension
-    //   on upper bound for 2D SBCC cases, and even should not limit to pow
-    //   of 2.
-
-    // FIXME: use all SBCC kernels instead after we fix the bugs in buffer assignment
-    // std::set<int> sbcc_support = {50, 64, 81, 100, 128, 200, 256};
-    std::set<int> sbcc_support = {64, 128, 256};
-    if((sbcc_support.find(length[1]) != sbcc_support.end()) && (length[0] >= 64))
+    try
+    {
```

Comment:
This looks like we should expect the try block to fail under normal circumstances.  Using exceptions for normal flow seems like a bad idea; exceptions should be for exceptional circumstances.

Reply:
Rearranging that should be the subject of a different PR, though.